tools/src/main/scala/scala/scalanative/codegen/Metadata.scala

```diff
@@ -5,7 +5,9 @@ import scala.collection.mutable
 import scalanative.nir._
 import scalanative.linker.{Trait, Class}
 
-class Metadata(val linked: linker.Result, proxies: Seq[Defn]) {
+class Metadata(val linked: linker.Result,
+               proxies: Seq[Defn],
+               val is32: Boolean) {
```

Comment:
Also if we would use linktime conditions we can get resolved information about is it 32-bit environment from `linked.resolvedVals(<name of proprty>)